src/main/java/teammates/storage/sqlentity/FeedbackSession.java

```diff
@@ -91,10 +93,12 @@ public class FeedbackSession extends BaseEntity {
 
     @OneToMany(mappedBy = "feedbackSession", cascade = CascadeType.REMOVE)
     @Fetch(FetchMode.JOIN)
+    @OnDelete(action = OnDeleteAction.CASCADE)
     private List<DeadlineExtension> deadlineExtensions = new ArrayList<>();
 
     @OneToMany(mappedBy = "feedbackSession", cascade = CascadeType.REMOVE)
     @Fetch(FetchMode.JOIN)
+    @OnDelete(action = OnDeleteAction.CASCADE)
```

Comment:
could you help to check if we need to add this for FeedbackResponse as well, for FeedbackResponseComments, currently it doesn't have @OnDelete, but i think it does cascade delete due to the @OneToMany, but I'm not a 100% sure

Reply:
Cedric, I have improved the IT and have checked `DataBundleLogicIT.json` to ensure that there is indeed, some FeedbackSession, FeedbackQuestion, FeedbackResponse, FeedbackResponseComment and DeadlineExtension linked to the `session1InTypicalCourse` course. I can confirm that the cascading works.

To be honest, it's a little wonky. I retried and removed `@OnDelete(action = OnDeleteAction.CASCADE)` from the entity classes that I have added and only removing `@OnDelete(action = OnDeleteAction.CASCADE)` from FeedbackQuestion results in the IT passing. The ones in FeedbackSession are required. i.e.,:

![image](https://github.com/TEAMMATES/teammates/assets/46486515/a35ed19b-52b5-4888-b188-e06fad1ab93c)

For clarity ([source](https://docs.jboss.org/hibernate/orm/6.4/userguide/html_single/Hibernate_User_Guide.html#pc-cascade-remove)):

> 6.15.3. CascadeType.REMOVE
The CascadeType.REMOVE allows us to remove a child entity along with the parent one. Traditionally, Hibernate called this operation delete, that’s why the org.hibernate.annotations.CascadeType provides a DELETE cascade option. However, CascadeType.REMOVE and org.hibernate.annotations.CascadeType.DELETE are identical.

Also for reference:
`@OnDelete(action = OnDeleteAction.CASCADE)` is used to specify the `ON DELETE CASCADE` action at the database level, ensuring that the database itself enforces the cascading delete behavior.